src/features/unified-share-modal/__tests__/UnifiedShareForm.test.js

```diff
@@ -211,14 +211,30 @@ describe('features/unified-share-modal/UnifiedShareForm', () => {
             expect(wrapper).toMatchSnapshot();
         });
 
-        test('should render a default component with upgrade CTA when showUpgradeOptions is enabled', () => {
-            const wrapper = getWrapper({
-                canInvite: true,
-                isFetching: false,
-                showUpgradeOptions: true,
-            });
-            expect(wrapper.exists('UpgradeBadge')).toBe(true);
-        });
+        test.each([
+            [true, true, null],
+            [true, false, null],
+            [false, true, null],
+            [false, false, null],
+            [true, true, mockUpsellInlineNotice],
+            [true, false, mockUpsellInlineNotice],
+            [false, true, mockUpsellInlineNotice],
+            [false, false, mockUpsellInlineNotice],
+        ])(
+            'should render a default component with upgrade CTA with respect to showUpgradeInlineNotice, showUpgradeOptions, and upsellInlineNotice props',
```

Comment:
We need to make this unit test description dynamic based on the conditions you listed in the array otherwise its going to generate 8 unit tests of the same name
refer to the jest documentation on your preferred way of making it dynamic - https://jestjs.io/docs/api#testeachtablename-fn-timeout